recipes/libcurl/all/conanfile.py

```diff
@@ -379,21 +387,26 @@ def _configure_autotools(self):
         return self._autotools, self._configure_autotools_vars()
 
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions['BUILD_TESTING'] = False
-        cmake.definitions['BUILD_CURL_EXE'] = False
-        cmake.definitions['CURL_DISABLE_LDAP'] = not self.options.with_ldap
-        cmake.definitions['BUILD_SHARED_LIBS'] = self.options.shared
-        cmake.definitions['CURL_STATICLIB'] = not self.options.shared
-        cmake.definitions['CMAKE_DEBUG_POSTFIX'] = ''
-        cmake.definitions['CMAKE_USE_LIBSSH2'] = self.options.with_libssh2
+        if self._cmake:
+            return self._cmake
+        if self._is_win_x_android:
+            self._cmake = CMake(self, generator="Ninja")
```

Comment:
I would add `ninja/1.9.0` to build requirements. Not optional if installed, but the package only.

Reply:
ping @a4z

Reply:
> Not optional if installed, but the package only.

not sure how I shall interpret this, can you please explain?
I will meanwhile start with adding ninja as a build requirement if there is a Android build on Windows running

Reply:
please check my latest commit, if this is what you meant , @uilianries

Reply:
@a4z Ninja or any other build requirement should be declared as explicit build requirement. Why? We have few cases where users can't build due a system package which could be a build requirement. So, let's say someone can't build curl for Android, will be harder to replicate the same scenario, as the user is running a system version. Okay, CMake and Compilers are exception.

Reply:
Is there a policy to use version ranges in build requirements btw?

Reply:
@Minimonium  I believe the same policy as with regular requirements.
e.g. using different version of nasm/yasm may result in different machine code generation
it doesn't affect on package id, but build is no longer truly reproducible in this case